src/mono/browser/runtime/loader/assets.ts

```diff
@@ -318,6 +318,11 @@ export function prepareAssets () {
         }
 
         const addAsset = (asset: AssetEntryInternal, isCore: boolean) => {
+            if (resources.fingerprinting && asset.behavior == "assembly") {
+                const indexOfExtension = asset.name.lastIndexOf(".");
+                const indexOfFingerprint = asset.name.lastIndexOf(".", indexOfExtension - 1);
+                asset.virtualPath = asset.name.substring(0, indexOfFingerprint) + asset.name.substring(indexOfExtension);
+            }
```

Comment:
Alternatively, for now, you could consider including this mapping in blazor.boot.json and compute it there so that you don't have to make the JS side aware of fingerprinting.

Reply:
At least corelib is matched by filename by the mono runtime.
Longer term plan (not sure about timing) is to update the boot config schema so we can attach similar metadata to assets. For now I prefer to do the math.